android/sdl_android/src/main/java/com/smartdevicelink/managers/file/FileManager.java

```diff
@@ -81,100 +70,33 @@ public FileManager(ISdl internalInterface, Context context, FileManagerConfig fi
+    InputStream openInputStreamWithFile(@NonNull SdlFile file) {
+        InputStream inputStream = null;
+
+        if (context.get() == null) {
+            DebugTool.logError(TAG, "Context is null. Cannot open file input stream!");
+            return null;
         }
 
         if (file.getResourceId() > 0) {
-            // Use resource id to upload file
-            byte[] contents = contentsOfResource(file.getResourceId());
-            if (contents != null) {
-                putFile.setFileData(contents);
-            } else {
-                throw new IllegalArgumentException("Resource file id was empty");
+            try {
+                inputStream = context.get().getResources().openRawResource(file.getResourceId());
+            } catch (Resources.NotFoundException e) {
+                DebugTool.logError(TAG, "File cannot be found.");
             }
         } else if (file.getUri() != null) {
-            // Use URI to upload file
-            byte[] contents = contentsOfUri(file.getUri());
-            if (contents != null) {
-                putFile.setFileData(contents);
-            } else {
-                throw new IllegalArgumentException("Uri was empty");
+            try {
+                inputStream = context.get().getContentResolver().openInputStream(file.getUri());
+            } catch (FileNotFoundException e) {
+                DebugTool.logError(TAG, String.format("File at %s cannot be found.", file.getUri()));
             }
         } else if (file.getFileData() != null) {
-            // Use file data (raw bytes) to upload file
-            putFile.setFileData(file.getFileData());
+            inputStream = new ByteArrayInputStream(file.getFileData());
         } else {
-            throw new IllegalArgumentException("The SdlFile to upload does " +
-                    "not specify its resourceId, Uri, or file data");
-        }
-
-        if (file.getType() != null) {
-            putFile.setFileType(file.getType());
-        }
-        putFile.setPersistentFile(file.isPersistent());
-
-        return putFile;
-    }
-
-    /**
-     * Helper method to take resource files and turn them into byte arrays
-     *
-     * @param resource Resource file id
-     * @return Resulting byte array
-     */
-    private byte[] contentsOfResource(int resource) {
-        InputStream is = null;
-        try {
-            is = context.get().getResources().openRawResource(resource);
-            return contentsOfInputStream(is);
-        } catch (Resources.NotFoundException e) {
-            DebugTool.logError(TAG, "Can't read from resource", e);
-            return null;
-        } finally {
-            if (is != null) {
-                try {
-                    is.close();
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-            }
+            DebugTool.logError(TAG, "The SdlFile to upload does not specify its resourceId, Uri, or file data");
```

Comment:
If the stream is null, it will return immediately in the method that called `openInputStreamWithFile()` :
https://github.com/smartdevicelink/sdl_java_suite/blob/2076d116208214d92424e2f652337ecab0104e16/base/src/main/java/com/smartdevicelink/managers/file/UploadFileOperation.java#L127

However, you are right, it can try again if the retry count is > 0 which is not ideal. But in my opinion, that is okay because it will realize quickly again in the second attempt that it cannot upload and return without executing a lot of unnecessary logic.

We can definitely improve that but it will complicate the logic and I don't think that it is a very common corner case. If you still be live that this should be improved, we can discuss that with the iOS team because both projects should align on how that case is handled.